From 0f81b49e797866968cf91049dd15114310c60c25 Mon Sep 17 00:00:00 2001
From: Simon Holesch <simon.holesch@bshg.com>
Date: Wed, 23 Feb 2022 16:01:34 +0100
Subject: [PATCH] Find drm_fourcc.h in sysroot

When PKG_CONFIG_SYSROOT_DIR is set, the pkg-config variable `includedir`
only contains the path inside the sysroot. To find the actual file,
prepend the sysroot (defaults to `/`).
---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 947a490..bfd02fe 100644
--- a/meson.build
+++ b/meson.build
@@ -62,7 +62,8 @@ if libdrm.version().version_compare('<2.4.104')
 elif libdrm.type_name() == 'internal'
   fourcc_h = files('subprojects/libdrm/include/drm/drm_fourcc.h')
 else
-  fourcc_h = files(libdrm.get_pkgconfig_variable('includedir') / 'libdrm/drm_fourcc.h')
+  fourcc_h = files(libdrm.get_pkgconfig_variable('pc_sysrootdir') +
+    libdrm.get_pkgconfig_variable('includedir') / 'libdrm/drm_fourcc.h')
 endif
 
 if libpci.found()
